File: app/Models/LeaveType.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class LeaveType extends Model
{
    use HasFactory;

    protected $fillable = [
        'name', 
        'slug', 
        'max_days_per_year', 
        'requires_advance_notice', 
        'advance_notice_days', 
        'enforce_advance_notice',
        'allows_retroactive', 
        'enforce_retroactive_check',
        'max_retroactive_days',
        'enforce_balance_check',
        'requires_file'
    ];

    protected $casts = [
        'requires_advance_notice' => 'boolean',
        'enforce_advance_notice' => 'boolean',
        'allows_retroactive' => 'boolean',
        'enforce_retroactive_check' => 'boolean',
        'enforce_balance_check' => 'boolean',
        'requires_file' => 'boolean',
    ];
}

File: nass_app/lib/models/leave_type.dart
class LeaveType {
  final int id;
  final String name;
  final String slug;
  final String? description;
  final int? maxDaysPerYear;
  final bool requiresAdvanceNotice;
  final int? advanceNoticeDays;
  final bool allowsRetroactive;

  LeaveType({
    required this.id,
    required this.name,
    required this.slug,
    this.description,
    this.maxDaysPerYear,
    this.requiresAdvanceNotice = false,
    this.advanceNoticeDays,
    this.allowsRetroactive = false,
  });

  factory LeaveType.fromJson(Map<String, dynamic> json) {
    return LeaveType(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      slug: json['slug'] ?? '',
      description: json['description'],
      maxDaysPerYear: json['max_days_per_year'],
      requiresAdvanceNotice: json['requires_advance_notice'] ?? false,
      advanceNoticeDays: json['advance_notice_days'],
      allowsRetroactive: json['allows_retroactive'] ?? false,
    );
  }
}
